cmd/cosign/cli/sign/sign.go

```diff
@@ -243,7 +243,11 @@ func signDigest(ctx context.Context, digest name.Digest, payload []byte, ko opti
 
 		s = tsa.NewSigner(s, clientTSA)
 	}
-	if ShouldUploadToTlog(ctx, ko, digest, ko.SkipConfirmation, tlogUpload) {
+
+	// Check if TSA signing is enabled and tlog upload is disabled
+	if !tlogUpload && ko.TSAServerURL != "" {
```

Comment:
could we put this check in `ShouldUploadToTlog` itself so that it's respected in all places that call that function?

Reply:
I wanted to wait until all the different parts using this function will have support for TSA. Otherwise it looks like adding a tsa parameter that is only used for images at the moment.

Reply:
It should still work though right, since tsaurl will never be set for the other commands? We'd be avoiding creating tech debt and it'll still work correctly.

Reply:
I don't think there is any tech debt yet, but this is something easy to solve for everyone.